preload.sh

```diff
@@ -13,76 +13,188 @@ API_HOST=${API_HOST:-"https://api.resin.io"}
 REGISTRY_HOST=${REGISTRY_HOST:-"registry.resin.io"}
 
 function cleanup {
-        rm $TMP_APPS_JSON || true
-	test "$DOCKER_PID" && kill $(cat $DOCKER_PID)
-	echo "Waiting for Docker to stop..."
-	while [ -e "$DOCKER_PID" ]; do
-		sleep 1
-	done
-        test -d "/tmp/docker-$APP_ID" && rm -rf "/tmp/docker-$APP_ID"
-        test "`mount | grep \"/mnt/$APP_ID\"`" && umount "/mnt/$APP_ID"
-        test -d "/mnt/$APP_ID" && rmdir "/mnt/$APP_ID"
+  test -e $TMP_APPS_JSON && rm $TMP_APPS_JSON || true
+  test "$DOCKER_PID" && kill $(cat $DOCKER_PID)
+  
+  echo "Waiting for Docker to stop..."
+  while [ -e "$DOCKER_PID" ]; do
+    sleep 1
+  done
+  
+  test -d "/tmp/docker-$APP_ID" && rm -rf "/tmp/docker-$APP_ID"
+  test "`mount | grep \"/mnt/$APP_ID\"`" && umount "/mnt/$APP_ID" || true
```

Comment:
Can you use `$()` instead of ``` `` ``` - I think that also avoids you needing to escape the inner `"` quotes